common/stats/disk_monitor.go

```diff
@@ -0,0 +1,97 @@
+package stats
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// DiskMonitor monitor disk usage for selected directories
+type DiskMonitor struct {
+	shortName  []string
+	paths      []string
+	startSizes []int64
+	endSizes   []int64
+}
+
+// NewDiskMonitor return a DiskMonitor
+func NewDiskMonitor(shortNames []string, paths []string) *DiskMonitor {
+	dm := &DiskMonitor{shortName: shortNames, paths: paths, startSizes: make([]int64, len(paths)), endSizes: make([]int64, len(paths))}
+	for i := range paths {
+		dm.startSizes[i] = -1
+		dm.endSizes[i] = -1
+	}
+	return dm
+}
+
+// GetStartSizes get the starting sizes of the directories being monitored
+func (dm *DiskMonitor) GetStartSizes() {
+	dm.getSizes(true)
+}
+
+// GetEndSizes get the ending sizes of the directories being monitored
+func (dm *DiskMonitor) GetEndSizes() {
+	dm.getSizes(false)
+}
+
+// RecordSizeStats record the disk size deltas to the stats receiver
+func (dm *DiskMonitor) RecordSizeStats(stat StatsReceiver) {
+	for i, shortName := range dm.shortName {
+		delta := dm.endSizes[i] - dm.startSizes[i]
+		statName := fmt.Sprintf("%s_%s", WorkerDirSizeChange, shortName)
+		stat.Gauge(statName).Update(delta)
+	}
+}
+
+// getStartSizes get the starting sized of the directories being monitored
+func (dm *DiskMonitor) getSizes(start bool) {
```

Comment:
nit: i think "isStart" is clearer than just "start"

Reply:
agreed, changed.